src/model_repository_manager.cc

```diff
@@ -364,10 +364,12 @@ ModelRepositoryManager::Create(
     std::unordered_map<std::string, std::vector<const InferenceParameter*>>
         models;
     for (const auto& model_name : startup_models) {
+      // Verify each model loads successfully, one at a time
       models[model_name];
+      RETURN_IF_ERROR(
+          local_manager->LoadUnloadModel(models, ActionType::LOAD, false));
+      models.clear();
     }
-    RETURN_IF_ERROR(local_manager->LoadUnloadModels(
-        models, ActionType::LOAD, false, &all_models_polled));
```

Comment:
for polling use case, all_models_polled will be updated as normal.

For explicitly loading startup models, it will instead fail immediately if any models fails to load, so the all_models_polled check is not necessary for startup models. And since `all_models_polled` defaults to `true` it should proceed as normal on success.